packages/node_modules/@webex/plugin-meetings/src/common/errors/webex-meetings-error.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * @extends Error
+ * @property {number} code - Error code
+ */
+export default class WebexError extends Error {
+  /**
+   * Creates a new {@link WebexMeetingsError}
+   * @param {number} code - Error code
+   * @param {string} [message] - Error message
+   * @param {string} [fileName] - Name of the script file where error was generated
+   * @param {number} [lineNumber] - Line number of the script file where error was generated
+   */
+  constructor(code) {
+    // eslint-disable-next-line prefer-rest-params
+    const args = [].slice.call(arguments, 1);
+
+    super(...args);
+
+    const error = Error.apply(this, args);
+
+    error.name = 'WebexError';
+
+    Object.defineProperty(this, 'code', {
+      value: code,
+      enumerable: true
+    });
+
+    Object.getOwnPropertyNames(error).forEach(function setDefaultProperty(prop) {
+      Object.defineProperty(this, prop, {
+        value: error[prop],
+        enumerable: true
+      });
+    }, this);
```

Comment:
Why are you setting these properties three times?
`super(...args);` is setting the Error object from which we are inheriting.
`const error = Error.apply(this, args);` sets the Error object again and getting a pointer to it.
`Object.getOwnPropertyNames.....` is setting the Error object for a third time.

Reply:
@mwegman i tried using it from twilio , I could be wrong but 
1) super(...args) and const error = Error.apply(this, args); does the same thing i guess the second line was ment to get the error instance and change the name 

2) getOwnPropertyNames takes the main error object and sets the values in the WebExError